### PR TITLE
Customising branding

### PR DESCRIPTION
**Pull request by adam-code-labx**
Reference to #458 
This PR includes the following:

- The logo and logo icon have now been extracted to a component and views now publishable
- Favicon can be overridden in the system config (perhaps better to extract to views/components/branding?)
- GetCandy shows clearly in the footer with the small stamp icon (Note: version should not show the word beta)
- Version set in the system config (We need to perhaps find away to automate this for releases?)
- Login now updated to use app name as the welcome message
- Login and password reset now use the new published logo component
- Lang now published

@glennjacobs @alecritson  
I hope these modifications are acceptable if you need anything else added please let me know 👍

### CODE REVIEW

**Comment by adam-code-labx** [comment]
@alecritson with the latest commit you can now extend scripts and styles outside the hub via stacks.
This will be useful for add-ons and to be able to have more control from the application your building.

Example usage:

```php
Event::listen(fn (RouteMatched $event) => $this->extendHubAssets($event));

/**
 * Extend Hub Assets
 *
 * @param  \Illuminate\Routing\Events\RouteMatched  $event
 * @return void
 */
protected function extendHubAssets(RouteMatched $event): void
{
    if (! str_starts_with($event->route->getName(), 'hub.')) {
        return;
    }

    app('view')->startPush('hub-styles', collect([
        '[data-ref=layout-sidemenu] { background: rgba(2, 132, 199, 0.1); };',
    ])->map(fn ($style) => "<style>{$style}</style>")->implode("\n"));

    app('view')->startPush('hub-scripts', collect([
        'console.info(\'Extend hub scripts\');',
        'Livewire.hook(\'component.initialized\', (component) => console.info(component.name));',
    ])->map(fn ($script) => "<script>{$script}</script>")->implode("\n"));
}
```

We could extend on this to perhaps add HubAssets manifest which would be cleaner.
I understand as you don't want to support theming via the hub, it should be enough for developers to handle this as long as we the styles & scripts @stack

**Comment by adam-code-labx** [comment]
In the latest commit added support for vite. 
So If devs are using the latest Laravel 9 with vite it is even simpler to customise the hub with full tailwind support.

Example.

```php
View::composer('adminhub::*', function (\Illuminate\View\View $view) {
    $view->with([
        'viteSupport' => Blade::defaultAliases()->keys()->contains('Vite'),
        'vite' => ['resources/css/hub-extend.css', 'resources/js/hub-extend.js'],
    ]);
});
```

**Comment by glennjacobs** [comment]
> @alecritson with the latest commit you can now extend scripts and styles outside the hub via stacks. This will be useful for add-ons and to be able to have more control from the application your building.
> 
> Example usage:
> 
> ```
> Event::listen(fn (RouteMatched $event) => $this->extendHubAssets($event));
> 
> /**
>  * Extend Hub Assets
>  *
>  * @param  \Illuminate\Routing\Events\RouteMatched  $event
>  * @return void
>  */
> protected function extendHubAssets(RouteMatched $event): void
> {
>     if (! str_starts_with($event->route->getName(), 'hub.')) {
>         return;
>     }
> 
>     app('view')->startPush('hub-styles', collect([
>         '[data-ref=layout-sidemenu] { background: rgba(2, 132, 199, 0.1); };',
>     ])->map(fn ($style) => "<style>{$style}</style>")->implode("\n"));
> 
>     app('view')->startPush('hub-scripts', collect([
>         'console.info(\'Extend hub scripts\');',
>         'Livewire.hook(\'component.initialized\', (component) => console.info(component.name));',
>     ])->map(fn ($script) => "<script>{$script}</script>")->implode("\n"));
> }
> ```
> 
> We could extend on this to perhaps add HubAssets manifest which would be cleaner. I understand as you don't want to support theming via the hub, it should be enough for developers to handle this as long as we the styles & scripts stacks

This is already handled by #279

**Comment by adam-code-labx** [comment]
> > @alecritson with the latest commit you can now extend scripts and styles outside the hub via stacks. This will be useful for add-ons and to be able to have more control from the application your building.
> > Example usage:
> > ```
> > Event::listen(fn (RouteMatched $event) => $this->extendHubAssets($event));
> > 
> > /**
> >  * Extend Hub Assets
> >  *
> >  * @param  \Illuminate\Routing\Events\RouteMatched  $event
> >  * @return void
> >  */
> > protected function extendHubAssets(RouteMatched $event): void
> > {
> >     if (! str_starts_with($event->route->getName(), 'hub.')) {
> >         return;
> >     }
> > 
> >     app('view')->startPush('hub-styles', collect([
> >         '[data-ref=layout-sidemenu] { background: rgba(2, 132, 199, 0.1); };',
> >     ])->map(fn ($style) => "<style>{$style}</style>")->implode("\n"));
> > 
> >     app('view')->startPush('hub-scripts', collect([
> >         'console.info(\'Extend hub scripts\');',
> >         'Livewire.hook(\'component.initialized\', (component) => console.info(component.name));',
> >     ])->map(fn ($script) => "<script>{$script}</script>")->implode("\n"));
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We could extend on this to perhaps add HubAssets manifest which would be cleaner. I understand as you don't want to support theming via the hub, it should be enough for developers to handle this as long as we the styles & scripts stacks
> 
> This is already handled by #279

Looks good and is this working outside the hub for example in AppServiceProvider? If so I agree we can remove stacks but should keep vite for better tailwind support. https://laravel.com/docs/9.x/vite#loading-your-scripts-and-styles

File: packages/admin/config/system.php [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?php
 
 return [
+    'version' => 'v2.0.15',
```
Comment (alecritson) [inline packages/admin/config/system.php:4]: This shouldn't be hard coded, it's a maintenance overhead and will quickly be outdated.

File: packages/admin/resources/views/layouts/app.blade.php [hunk 2]
```diff
@@ -40,7 +38,12 @@ class="h-full">
         })
     </script>
 
+    @if ($viteSupport ?? false)
```
Comment (alecritson) [inline packages/admin/resources/views/layouts/app.blade.php:41]: Not sure why there is Vite logic in here? It's not been spoke about / agreed and potentially conflicts with another PR (as @glennjacobs mentioned) this needs to go.

File: packages/admin/resources/views/layouts/app.blade.php [hunk 3]
```diff
@@ -40,7 +38,12 @@ class="h-full">
...
+        @vite($vite ?? [])
+    @endif
+
     @livewireStyles
+    @stack('hub-styles')
```
Comment (alecritson) [inline packages/admin/resources/views/layouts/app.blade.php:46]: Again, potential conflict / not spoken about, needs to go.

File: packages/admin/resources/views/partials/navigation/side-menu.blade.php [hunk 4]
```diff
@@ -7,25 +7,16 @@
...
+           <x-hub::branding.logo x-show="!showExpandedMenu" :icon-only="true" />
         </a>
 
         <div class="w-full px-4 pt-4 border-t border-gray-100 dark:border-gray-800">
             @livewire('sidebar')
         </div>
 
         <button x-on:click="showExpandedMenu = !showExpandedMenu"
-                class="absolute z-50 p-1 bg-white border border-gray-100 dark:border-gray-700 dark:bg-gray-800 rounded -right-[13px] bottom-8 text-gray-600 hover:text-gray-700 dark:text-gray-400 dark:hover:text-gray-300">
+                class="absolute z-50 p-1 bg-white border border-gray-100 dark:border-gray-700 dark:bg-gray-800 rounded -right-[13px] bottom-12 text-gray-600 hover:text-gray-700 dark:text-gray-400 dark:hover:text-gray-300">
```
Comment (alecritson) [inline packages/admin/resources/views/partials/navigation/side-menu.blade.php:19]: What's the reasoning for this change? @markmead

**Review by alecritson — COMMENTED** [review]
Thanks @adam-code-labx There's a few things in the PR that would need addressing, howver, as mentioned in the referenced issue, this was something we'd likely want to look at ourselves, I would say don't do too much more on this in case we want to go in another direction.

File: packages/admin/resources/views/partials/navigation/side-menu.blade.php (hunk 4 above)
Comment (markmead) [reply packages/admin/resources/views/partials/navigation/side-menu.blade.php:19]: I'm not too sure... Was there an issue @adam-code-labx?

File: packages/admin/config/system.php (hunk 1 above)
Comment (adam-code-labx) [reply packages/admin/config/system.php:4]: Should perhaps be defined in a class like Laravel does with their versioning

File: packages/admin/resources/views/layouts/app.blade.php (hunk 2 above)
Comment (adam-code-labx) [reply packages/admin/resources/views/layouts/app.blade.php:41]: Stacks could conflict but vite will not very useful as it auto compiles tailwind. Also viteSupports checks if vite exists on the Laravel install

File: packages/admin/resources/views/layouts/app.blade.php (hunk 3 above)
Comment (adam-code-labx) [reply packages/admin/resources/views/layouts/app.blade.php:46]: Ok

File: packages/admin/resources/views/partials/navigation/side-menu.blade.php (hunk 4 above)
Comment (adam-code-labx) [reply packages/admin/resources/views/partials/navigation/side-menu.blade.php:19]: Due to the new footer

**Comment by adam-code-labx** [comment]
@alecritson Removed last 2 commits
Will leave this up to you guys now on this one, hopefully what I have done is a starting point at least 😉

File: packages/admin/config/system.php (hunk 1 above)
Comment (alecritson) [reply packages/admin/config/system.php:4]: I think this should come from what's installed in composer personally.

**Comment by alecritson** [comment]
Thanks @adam-code-labx I've pulled this down and had a look.

I think for the login page we should allow the full logo to be displayed, not just the icon version. We have enough space there to handle it.

I've not sure about the footer, creates a white bar across the site, I though the version info would just be at the bottom of the pages in the grey area as below.

image

**Comment by glennjacobs** [comment]
@alecritson Would it make sense for @markmead to make the changes we want? Otherwise, we may just go round in circles.

**Comment by alecritson** [comment]
@glennjacobs Yeah if we think that's the best place for it.

File: packages/admin/config/system.php (hunk 1 above)
Comment (adam-code-labx) [reply packages/admin/config/system.php:4]: @alecritson 
Sure you can use this which outputs "2.0-beta15 `Composer\InstalledVersions::getPrettyVersion('getcandy/admin')`

Can I suggest we please at least str_replace to remove beta? 2.0.15

**Comment by adam-code-labx** [comment]
I don't mind doing those modifications otherwise @markmead can handle this one up to you guys 😉

**Comment by glennjacobs** [comment]
@markmead is going to work on this today so we can hopefully get it merged.

**Review by markmead — APPROVED** [review]

**Pull request merged**